src/ORM/Association/HasMany.php

```diff
@@ -134,6 +185,47 @@ public function saveAssociated(EntityInterface $entity, array $options = [])
     }
 
     /**
+     * Deletes/sets null the related objects according to the dependency between source and targets and foreign key nullability
+     *
+     * @param array $properties array of foreignKey properties
+     * @param EntityInterface $entity the entity which should have its associated entities unassigned
+     * @param Table $target The associated table
+     * @param array $options original list of options passed in constructor
+     * @return void
+     */
+    protected function _unlinkAssociated(array $properties, EntityInterface $entity, Table $target, array $options)
+    {
+        $mustBeDependent = (!$this->_foreignKeyAcceptsNull($target, $properties) || $this->dependent());
+        $this->dependent($mustBeDependent);
+
+        if ($mustBeDependent) {
+                $this->cascadeDelete($entity, $options);
+        } else {
+            $updateFields = array_fill_keys(array_keys($properties), null);
+            $target->updateAll($updateFields, $properties);
+        }
+    }
+
+    /**
+     * Checks the nullable flag of the foreign key
+     *
+     * @param Table $table the table containing the foreign key
+     * @param array $properties the list of fields that compose the foreign key
+     * @return bool
+     */
+    protected function _foreignKeyAcceptsNull(Table $table, array $properties)
+    {
+        return array_product(
```

Comment:
Once its domain as a number will be 0 and 1, I don't see much sense in having an int return value, so I will convert it to boolean. Thanks!